packages/dashboard-core-plugins/src/panels/NotebookPanel.tsx

```diff
@@ -959,6 +975,27 @@ class NotebookPanel extends Component<NotebookPanelProps, NotebookPanelState> {
     this.saveContent(name, content);
   }
 
+  canFormat(): boolean {
+    return (
+      this.notebook?.editor
+        ?.getAction('editor.action.formatDocument')
+        ?.isSupported() === true
+    );
+  }
+
+  async handleFormat(): Promise<void> {
+    if (this.canFormat()) {
+      await this.notebook?.editor
+        ?.getAction('editor.action.formatDocument')
+        ?.run();
+    }
+  }
+
+  async handleSaveFromShortcut(): Promise<void> {
+    await this.handleFormat();
```

Comment:
I don't think we should auto format on save by default unless the user has opted in, configured the formatting settings, or has otherwise been notified this will be occurring. Imposing our preferred formatting settings on others without their approval is likely to annoy people.
I know this is going into a feature branch, but I think important to call that out.

Reply:
I commented this out for now and added a TODO comment to check if the user enabled format on save.

You can still format from the overflow menu on this branch